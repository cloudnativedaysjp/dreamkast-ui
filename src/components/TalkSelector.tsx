import React from 'react'
//import { makeStyles } from '@material-ui/core/styles';
import { Talk } from '../client-axios'
import { List } from '@material-ui/core'
import { ListItem } from '@material-ui/core'
import { ListItemText } from '@material-ui/core'

type Props = {
  selectedTrackId: number
  selectedTalk?: Talk
  talks: Talk[]
  selectTalk: (talk: Talk) => void
}

// const useStyles = makeStyles({
//   player: {
//     //backgroundColor: "#CC0",
//   },
//   chat: {
//     //backgroundColor: "#CC0",
//   }
// });

const TalkSelector: React.FC<Props> = ({
  selectedTrackId,
  selectedTalk,
  talks,
  selectTalk,
}) => {
  // const classes = useStyles();

  return (
    <List component="nav">
      {talks.map((talk) => {
        if (talk.trackId == selectedTrackId) {
          return (
            <ListItem
              button
              key={talk.id}
              selected={talk.id === selectedTalk?.id}
              onClick={() => selectTalk(talk)}
            >
              <ListItemText inset primary={talk.title} />
            </ListItem>
          )
        }
      })}
    </List>
  )
}

export default TalkSelector
