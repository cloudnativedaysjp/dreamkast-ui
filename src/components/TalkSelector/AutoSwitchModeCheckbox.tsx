import React from 'react'
import {
  isAutoSwitchModeSelector,
  setIsAutoSwitchMode,
} from '../../store/settings'
import { useDispatch, useSelector } from 'react-redux'
import { Checkbox } from '@material-ui/core'

export const AutoSwitchModeCheckbox = () => {
  const dispatch = useDispatch()
  const isAutoSwitchMode = useSelector(isAutoSwitchModeSelector)
  const changeAutoSwitchMode = (checked: boolean) => {
    dispatch(setIsAutoSwitchMode(checked))
  }

  return (
    <Checkbox
      size="small"
      checked={isAutoSwitchMode}
      onChange={(_, checked) => changeAutoSwitchMode(checked)}
    />
  )
}
