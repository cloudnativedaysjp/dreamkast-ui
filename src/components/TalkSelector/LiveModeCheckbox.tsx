import React from 'react'
import {
  isLiveModeSelector,
  setIsAutoSwitchMode,
  setIsLiveMode,
} from '../../store/settings'
import { useDispatch, useSelector } from 'react-redux'
import { Checkbox } from '@material-ui/core'

export const LiveModeCheckbox = () => {
  const dispatch = useDispatch()
  const isLiveMode = useSelector(isLiveModeSelector)
  const changeLiveMode = (checked: boolean) => {
    if (!checked) {
      dispatch(setIsAutoSwitchMode(checked))
    }
    dispatch(setIsLiveMode(checked))
  }

  return (
    <Checkbox
      size="small"
      checked={isLiveMode}
      onChange={(_, checked) => changeLiveMode(checked)}
    />
  )
}
