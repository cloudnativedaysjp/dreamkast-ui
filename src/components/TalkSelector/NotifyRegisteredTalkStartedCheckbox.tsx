import React from 'react'
import {
  notifyRegisteredTalkStartedSelector,
  setNotifyRegisteredTalkStarted,
} from '../../store/settings'
import { useDispatch, useSelector } from 'react-redux'
import { Checkbox } from '@material-ui/core'

export const NotifyRegisteredTalkStartedCheckbox = () => {
  const dispatch = useDispatch()
  const isAutoSwitchMode = useSelector(notifyRegisteredTalkStartedSelector)
  const changeAutoSwitchMode = (checked: boolean) => {
    dispatch(setNotifyRegisteredTalkStarted(checked))
  }

  return (
    <Checkbox
      size="small"
      checked={isAutoSwitchMode}
      onChange={(_, checked) => changeAutoSwitchMode(checked)}
    />
  )
}
