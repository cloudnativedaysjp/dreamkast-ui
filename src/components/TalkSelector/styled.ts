import styled from 'styled-components'
import { List as OriginList, ListItem, ListItemText } from '@material-ui/core'

export const Container = styled.div`
  padding: 8px;
  height: 500px;
`

export const Title = styled.div`
  font-size: 16px;
  font-weight: bold;
  text-align: center;
  padding: 10px;
  border: 1.5px solid black;
  border-bottom: none;
  border-radius: 10px 10px 0px 0px;
  background-color: #ffffff;
  color: #037f8c;
`

export const List = styled(OriginList)`
  height: 450px;
  border: 1.5px solid;
  overflow-y: scroll;
  border-radius: 0px 0px 10px 10px;
  background-color: #ffffff;
  &::-webkit-scrollbar {
    display: none;
  }
`

export const Item = styled(ListItem)``

export const Text = styled(ListItemText)``
