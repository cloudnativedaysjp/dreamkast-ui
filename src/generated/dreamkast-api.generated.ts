import { baseApi as api } from '../store/baseApi'
export const addTagTypes = [
  'Profile',
  'Event',
  'Track',
  'Talk',
  'VideoRegistration',
  'ChatMessage',
  'Sponsor',
  'Booth',
] as const
const injectedRtkApi = api
  .enhanceEndpoints({
    addTagTypes,
  })
  .injectEndpoints({
    endpoints: (build) => ({
      getApiV1ByEventAbbrMyProfile: build.query<
        GetApiV1ByEventAbbrMyProfileApiResponse,
        GetApiV1ByEventAbbrMyProfileApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/$${queryArg.eventAbbr}/my_profile`,
        }),
        providesTags: ['Profile'],
      }),
      getApiV1Events: build.query<
        GetApiV1EventsApiResponse,
        GetApiV1EventsApiArg
      >({
        query: () => ({ url: `/api/v1/events` }),
        providesTags: ['Event'],
      }),
      getApiV1EventsByEventAbbr: build.query<
        GetApiV1EventsByEventAbbrApiResponse,
        GetApiV1EventsByEventAbbrApiArg
      >({
        query: (queryArg) => ({ url: `/api/v1/events/${queryArg.eventAbbr}` }),
        providesTags: ['Event'],
      }),
      getApiV1Tracks: build.query<
        GetApiV1TracksApiResponse,
        GetApiV1TracksApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/tracks`,
          params: { eventAbbr: queryArg.eventAbbr },
        }),
        providesTags: ['Track'],
      }),
      getApiV1TracksByTrackId: build.query<
        GetApiV1TracksByTrackIdApiResponse,
        GetApiV1TracksByTrackIdApiArg
      >({
        query: (queryArg) => ({ url: `/api/v1/tracks/${queryArg.trackId}` }),
        providesTags: ['Track'],
      }),
      getApiV1TracksByTrackIdViewerCount: build.query<
        GetApiV1TracksByTrackIdViewerCountApiResponse,
        GetApiV1TracksByTrackIdViewerCountApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/tracks/${queryArg.trackId}/viewer_count`,
        }),
        providesTags: ['Track'],
      }),
      getApiV1Talks: build.query<GetApiV1TalksApiResponse, GetApiV1TalksApiArg>(
        {
          query: (queryArg) => ({
            url: `/api/v1/talks`,
            params: {
              eventAbbr: queryArg.eventAbbr,
              trackId: queryArg.trackId,
              conferenceDayIds: queryArg.conferenceDayIds,
            },
          }),
          providesTags: ['Talk'],
        },
      ),
      getApiV1TalksByTalkId: build.query<
        GetApiV1TalksByTalkIdApiResponse,
        GetApiV1TalksByTalkIdApiArg
      >({
        query: (queryArg) => ({ url: `/api/v1/talks/${queryArg.talkId}` }),
        providesTags: ['Talk'],
      }),
      putApiV1TalksByTalkId: build.mutation<
        PutApiV1TalksByTalkIdApiResponse,
        PutApiV1TalksByTalkIdApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/talks/${queryArg.talkId}`,
          method: 'PUT',
          body: queryArg.body,
        }),
        invalidatesTags: ['Talk'],
      }),
      getApiV1TalksByTalkIdVideoRegistration: build.query<
        GetApiV1TalksByTalkIdVideoRegistrationApiResponse,
        GetApiV1TalksByTalkIdVideoRegistrationApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/talks/${queryArg.talkId}/video_registration`,
        }),
        providesTags: ['VideoRegistration'],
      }),
      putApiV1TalksByTalkIdVideoRegistration: build.mutation<
        PutApiV1TalksByTalkIdVideoRegistrationApiResponse,
        PutApiV1TalksByTalkIdVideoRegistrationApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/talks/${queryArg.talkId}/video_registration`,
          method: 'PUT',
          body: queryArg.videoRegistration,
        }),
        invalidatesTags: ['VideoRegistration'],
      }),
      getApiV1ChatMessages: build.query<
        GetApiV1ChatMessagesApiResponse,
        GetApiV1ChatMessagesApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/chat_messages`,
          params: {
            eventAbbr: queryArg.eventAbbr,
            roomId: queryArg.roomId,
            roomType: queryArg.roomType,
            createdFrom: queryArg.createdFrom,
          },
        }),
        providesTags: ['ChatMessage'],
      }),
      postApiV1ChatMessages: build.mutation<
        PostApiV1ChatMessagesApiResponse,
        PostApiV1ChatMessagesApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/chat_messages`,
          method: 'POST',
          body: queryArg.chatMessage,
        }),
        invalidatesTags: ['ChatMessage'],
      }),
      putApiV1ChatMessagesByMessageId: build.mutation<
        PutApiV1ChatMessagesByMessageIdApiResponse,
        PutApiV1ChatMessagesByMessageIdApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/chat_messages/${queryArg.messageId}`,
          method: 'PUT',
          body: queryArg.updateChatMessage,
        }),
        invalidatesTags: ['ChatMessage'],
      }),
      getApiV1Sponsors: build.query<
        GetApiV1SponsorsApiResponse,
        GetApiV1SponsorsApiArg
      >({
        query: (queryArg) => ({
          url: `/api/v1/sponsors`,
          params: { eventAbbr: queryArg.eventAbbr },
        }),
        providesTags: ['Sponsor'],
      }),
      getApiV1BoothsByBoothId: build.query<
        GetApiV1BoothsByBoothIdApiResponse,
        GetApiV1BoothsByBoothIdApiArg
      >({
        query: (queryArg) => ({ url: `/api/v1/booths/${queryArg.boothId}` }),
        providesTags: ['Booth'],
      }),
    }),
    overrideExisting: false,
  })
export { injectedRtkApi as dreamkastApi }
export type GetApiV1ByEventAbbrMyProfileApiResponse =
  /** status 200 OK */ Profile
export type GetApiV1ByEventAbbrMyProfileApiArg = {
  /** ID of event */
  eventAbbr: string
}
export type GetApiV1EventsApiResponse = /** status 200 OK */ Event[]
export type GetApiV1EventsApiArg = void
export type GetApiV1EventsByEventAbbrApiResponse = /** status 200 OK */ Event
export type GetApiV1EventsByEventAbbrApiArg = {
  /** ID of event */
  eventAbbr: string
}
export type GetApiV1TracksApiResponse = /** status 200 OK */ Track[]
export type GetApiV1TracksApiArg = {
  /** abbr of event (e.g. cndt2020) */
  eventAbbr: string
}
export type GetApiV1TracksByTrackIdApiResponse = /** status 200 OK */ Track
export type GetApiV1TracksByTrackIdApiArg = {
  /** ID of track */
  trackId: string
}
export type GetApiV1TracksByTrackIdViewerCountApiResponse =
  /** status 200 OK */ ViewerCount
export type GetApiV1TracksByTrackIdViewerCountApiArg = {
  /** ID of track */
  trackId: string
}
export type GetApiV1TalksApiResponse = /** status 200 OK */ Talk[]
export type GetApiV1TalksApiArg = {
  /** abbr of event (e.g. cndt2020) */
  eventAbbr: string
  /** ID of track */
  trackId?: string
  conferenceDayIds?: string
}
export type GetApiV1TalksByTalkIdApiResponse = /** status 200 OK */ Talk
export type GetApiV1TalksByTalkIdApiArg = {
  /** ID of talk */
  talkId: string
}
export type PutApiV1TalksByTalkIdApiResponse = unknown
export type PutApiV1TalksByTalkIdApiArg = {
  /** ID of talk */
  talkId: string
  /** Update on_air status */
  body: {
    on_air?: boolean | undefined
  }
}
export type GetApiV1TalksByTalkIdVideoRegistrationApiResponse =
  /** status 200 OK */ VideoRegistration
export type GetApiV1TalksByTalkIdVideoRegistrationApiArg = {
  /** ID of talk */
  talkId: string
}
export type PutApiV1TalksByTalkIdVideoRegistrationApiResponse = unknown
export type PutApiV1TalksByTalkIdVideoRegistrationApiArg = {
  /** ID of talk */
  talkId: string
  /** Update video_registration */
  videoRegistration: VideoRegistration
}
export type GetApiV1ChatMessagesApiResponse = /** status 200 OK */ ChatMessage[]
export type GetApiV1ChatMessagesApiArg = {
  /** abbr of event (e.g. cndt2020) */
  eventAbbr: string
  /** ID of chat room */
  roomId: string
  /** Type of chat room */
  roomType: string
  /** YYYY-MM-DDThh:mm:ss+TZD (use UTC) */
  createdFrom?: string
}
export type PostApiV1ChatMessagesApiResponse = unknown
export type PostApiV1ChatMessagesApiArg = {
  /** chat message to create */
  chatMessage: ChatMessage
}
export type PutApiV1ChatMessagesByMessageIdApiResponse =
  /** status 200 OK */ ChatMessage[]
export type PutApiV1ChatMessagesByMessageIdApiArg = {
  /** ID of ChatMessage */
  messageId: string
  /** chat message to update */
  updateChatMessage: UpdateChatMessage
}
export type GetApiV1SponsorsApiResponse = /** status 200 OK */ Sponsor[]
export type GetApiV1SponsorsApiArg = {
  /** abbr of event (e.g. cndt2020) */
  eventAbbr: string
}
export type GetApiV1BoothsByBoothIdApiResponse = /** status 200 OK */ Booth
export type GetApiV1BoothsByBoothIdApiArg = {
  /** ID of booth */
  boothId: string
}
export type Profile = {
  id: number
  name: string
}
export type Event = {
  id: number
  name: string
  abbr: string
  status: string
  theme: string
  about: string
  privacy_policy: string
  privacy_policy_for_speaker: string
  copyright: string
  coc: string
  conferenceDays?:
    | {
        id?: number | undefined
        date?: string | undefined
        internal?: boolean | undefined
      }[]
    | undefined
}
export type Track = {
  id: number
  name: string
  videoPlatform?: (string | null) | undefined
  videoId?: (string | null) | undefined
  channelArn?: (string | null) | undefined
  onAirTalk?: (object | null) | undefined
}
export type ViewerCount = {
  conference_id?: number | undefined
  track_id: number
  talk_id?: number | undefined
  viewer_count: number
}
export type Talk = {
  id: number
  conferenceId?: number | undefined
  trackId: number
  videoPlatform?: string | undefined
  videoId: string
  title: string
  abstract: string
  speakers: {
    id?: number | undefined
    name?: string | undefined
  }[]
  dayId: number | null
  showOnTimetable: boolean
  startTime: string
  endTime: string
  talkDuration: number
  talkDifficulty: string
  talkCategory: string
  onAir?: boolean | undefined
  documentUrl?: string | undefined
  conferenceDayId?: (number | null) | undefined
  conferenceDayDate?: (string | null) | undefined
  startOffset?: number | undefined
  endOffset?: number | undefined
  actualStartTime?: string | undefined
  actualEndTime?: string | undefined
  presentationMethod?: (string | null) | undefined
}
export type VideoRegistration = {
  url?: string | undefined
  status: 'unsubmitted' | 'submitted' | 'confirmed' | 'invalid_format'
  statistics?: object | undefined
  createdAt?: string | undefined
  updatedAt?: string | undefined
}
export type ChatMessageProperties = {
  id?: number | undefined
  profileId?: number | undefined
  speakerId?: (number | null) | undefined
  eventAbbr?: string | undefined
  roomId?: number | undefined
  roomType?: string | undefined
  body?: string | undefined
  createdAt?: string | undefined
  messageType?: ('chat' | 'qa') | undefined
  replyTo?: (number | null) | undefined
}
export type ChatMessage = ChatMessageProperties
export type UpdateChatMessage = ChatMessageProperties
export type Sponsor = {
  id: number
  eventAbbr: string
  name: string
  abbr: string
  url: string
  logo_url: string
  sponsorType?: string[] | undefined
  booth?:
    | {
        id?: number | undefined
        opened?: boolean | undefined
      }
    | undefined
}
export type Booth = {
  id: number
  sponsorId: number
  sponsorName: string
  published: boolean
  description: string
  url?: string | undefined
  abbr: string
  text: string
  logoUrl: string
  vimeoUrl: string
  miroUrl: string
  pdfUrls: {
    url?: string | undefined
    title?: string | undefined
  }[]
  keyImageUrls: string[]
}
export const {
  useGetApiV1ByEventAbbrMyProfileQuery,
  useGetApiV1EventsQuery,
  useGetApiV1EventsByEventAbbrQuery,
  useGetApiV1TracksQuery,
  useGetApiV1TracksByTrackIdQuery,
  useGetApiV1TracksByTrackIdViewerCountQuery,
  useGetApiV1TalksQuery,
  useGetApiV1TalksByTalkIdQuery,
  usePutApiV1TalksByTalkIdMutation,
  useGetApiV1TalksByTalkIdVideoRegistrationQuery,
  usePutApiV1TalksByTalkIdVideoRegistrationMutation,
  useGetApiV1ChatMessagesQuery,
  usePostApiV1ChatMessagesMutation,
  usePutApiV1ChatMessagesByMessageIdMutation,
  useGetApiV1SponsorsQuery,
  useGetApiV1BoothsByBoothIdQuery,
} = injectedRtkApi
