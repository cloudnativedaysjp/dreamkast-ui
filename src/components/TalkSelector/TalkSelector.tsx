import React, { ReactElement, useEffect, useMemo, useState } from 'react'
import * as Styled from './styled'
import dayjs from 'dayjs'
import { Talk } from '../../generated/dreamkast-api.generated'
import { setupDayjs } from '../../util/setupDayjs'
import { LiveModeCheckbox } from './LiveModeCheckbox'
import { NotifyRegisteredTalkStartedCheckbox } from './NotifyRegisteredTalkStartedCheckbox'

setupDayjs()

type Props = {
  selectedTrackId?: number
  selectedTalkId?: number
  talks: Talk[]
  selectTalk: (talkId: number) => void
  small?: boolean
}

export const TalkSelector: React.FC<Props> = (props) => {
  const [now, setNow] = useState<number>(dayjs().unix())

  useEffect(() => {
    if (!process.browser) return
    const id = window.setInterval(() => {
      setNow(dayjs().unix())
    }, 1000)
    return window.clearInterval(id)
  }, [])

  const footer = (
    <>
      <LiveModeCheckbox />
      <Styled.label>ライブセッションに自動遷移</Styled.label>
      <NotifyRegisteredTalkStartedCheckbox />
      <Styled.label>事前登録したセッションに自動遷移</Styled.label>
    </>
  )

  return <PTalkSelector {...props} now={now} footer={footer}></PTalkSelector>
}

type PProps = Props & {
  now: number
  footer?: ReactElement
}

export const PTalkSelector: React.FC<PProps> = ({
  selectedTrackId,
  selectedTalkId,
  talks,
  selectTalk,
  small = false,
  now,
  footer,
}) => {
  const availableTalks = useMemo(
    () => extractAvailableTalks(talks, now),
    [talks, now],
  )

  return (
    <Styled.Container>
      <Styled.Title>このトラックのセッション</Styled.Title>
      <Styled.List height={small ? 350 : 497}>
        {availableTalks.map((talk) => {
          if (talk.trackId == selectedTrackId) {
            return (
              <Styled.Item
                button
                key={talk.id}
                disabled={!talk.available}
                selected={talk.id === selectedTalkId}
                onClick={() => selectTalk(talk.id)}
              >
                <Styled.Text>
                  {talk.onAir && <Styled.Live>LIVE</Styled.Live>}{' '}
                  {dayjs(talk.startTime).tz().format('HH:mm')}-
                  {dayjs(talk.endTime).tz().format('HH:mm')}
                  <br />
                  {talk.title}
                </Styled.Text>
              </Styled.Item>
            )
          }
        })}
      </Styled.List>
      <Styled.Footer>{footer}</Styled.Footer>
    </Styled.Container>
  )
}

type TalkWithAvailable = Talk & {
  available: boolean
}

export const extractAvailableTalks = (
  talks: Talk[],
  now: number,
): TalkWithAvailable[] => {
  return [...talks]
    .sort((n1, n2) => dayjs(n1.startTime).unix() - dayjs(n2.startTime).unix())
    .filter((talk) => talk.showOnTimetable)
    .map((talk) => {
      return {
        ...talk,
        available: isAvailable(now, talk.startTime, talk.conferenceDayDate),
      }
    })
}

const isAvailable = (
  now: number,
  startTime: string,
  conferanceDayDate?: string | null,
) => {
  if (!conferanceDayDate) return true // okui: これであってる？
  const startDate = `${conferanceDayDate} ${dayjs(startTime).format('HH:mm')}`
  return now - dayjs(startDate).unix() >= 0
}
