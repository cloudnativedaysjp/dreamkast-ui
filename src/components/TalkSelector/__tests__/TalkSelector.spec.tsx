import React from 'react'
import {
  MockTalkA1,
  MockTalkA2,
  MockTalkA3,
  MockTalks,
} from '../../../testhelper/fixture'
import { extractAvailableTalks, PTalkSelector } from '../TalkSelector'
import { Talk } from '../../../generated/dreamkast-api.generated'
import dayjs from 'dayjs'
import { render } from '@testing-library/react'

describe('extractAvailableTalks', () => {
  it('should provide sorted talks with available props only for showOnTimetable flag is true', () => {
    const hide = (talk: Talk) => {
      talk.showOnTimetable = false
      return talk
    }

    const given = {
      talks: [
        MockTalkA2(),
        MockTalkA3(),
        MockTalkA1(),
        hide(MockTalkA1()),
        hide(MockTalkA2()),
      ],
      now: dayjs('2023-02-24T10:30:00.000+09:00').unix(),
    }
    const want: ReturnType<typeof extractAvailableTalks> = [
      { ...MockTalkA1(), available: true },
      { ...MockTalkA2(), available: true },
      { ...MockTalkA3(), available: false },
    ]
    expect(extractAvailableTalks(given.talks, given.now)).toStrictEqual(want)
  })
})

describe('PTalkSelector', () => {
  it('should render without crash', () => {
    const now = dayjs('2023-02-24T10:30:00.000+09:00').unix()
    const screen = render(
      <PTalkSelector
        selectedTalkId={MockTalks()[0].id}
        selectedTrackId={32}
        talks={MockTalks()}
        isLiveMode={true}
        changeLiveMode={() => null}
        selectTalk={() => null}
        now={now}
      />,
    )
    expect(screen.asFragment()).toMatchSnapshot()
  })
  it('should render when props undefined', () => {
    const now = dayjs('2023-02-24T10:30:00.000+09:00').unix()
    const screen = render(
      <PTalkSelector
        talks={[]}
        changeLiveMode={() => null}
        selectTalk={() => null}
        now={now}
      />,
    )
    expect(screen.asFragment()).toMatchSnapshot()
  })
})
