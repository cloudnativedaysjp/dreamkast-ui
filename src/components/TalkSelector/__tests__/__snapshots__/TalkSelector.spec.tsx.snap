// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PTalkSelector should render when props undefined 1`] = `
<DocumentFragment>
  <div
    class="styled__Container-sc-1piihja-0"
  >
    <div
      class="styled__Title-sc-1piihja-1 hrbxKr"
    >
      このトラックのセッション
    </div>
    <ul
      class="MuiList-root styled__List-sc-1piihja-2 TaJvQ MuiList-padding"
      height="497"
    />
    <div
      class="styled__Footer-sc-1piihja-6 cnpEUB"
    />
  </div>
</DocumentFragment>
`;

exports[`PTalkSelector should render without crash 1`] = `
<DocumentFragment>
  <div
    class="styled__Container-sc-1piihja-0"
  >
    <div
      class="styled__Title-sc-1piihja-1 hrbxKr"
    >
      このトラックのセッション
    </div>
    <ul
      class="MuiList-root styled__List-sc-1piihja-2 TaJvQ MuiList-padding"
      height="497"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root styled__Item-sc-1piihja-3 iyjOuh MuiListItem-gutters MuiListItem-button Mui-selected"
        role="button"
        tabindex="0"
      >
        <div
          class="styled__Text-sc-1piihja-4 dleXVB"
        >
          <span
            class="styled__Live-sc-1piihja-5 gBQmsg"
          >
            LIVE
          </span>
           10:00-10:20
          <br />
          Securityに関する発表
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root styled__Item-sc-1piihja-3 ifFYpr MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="styled__Text-sc-1piihja-4 dleXVB"
        >
           10:25-10:45
          <br />
          IoT/Edgeに関する10の知見
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="true"
        class="MuiButtonBase-root MuiListItem-root styled__Item-sc-1piihja-3 ifFYpr MuiListItem-gutters Mui-disabled MuiListItem-button Mui-disabled"
        role="button"
        tabindex="-1"
      >
        <div
          class="styled__Text-sc-1piihja-4 dleXVB"
        >
           10:50-11:10
          <br />
          CloudNative Networking
        </div>
      </div>
    </ul>
    <div
      class="styled__Footer-sc-1piihja-6 cnpEUB"
    />
  </div>
</DocumentFragment>
`;
